qsiprep/workflows/dwi/merge.py

```diff
@@ -329,14 +454,22 @@ def init_dwi_denoising_wf(
 
     def get_buffernode():
         num_buffers = len(buffernodes)
-        return pe.Node(niu.IdentityInterface(fields=["dwi_file"]), name="buffer%02d" % num_buffers)
+        return pe.Node(
+            niu.IdentityInterface(fields=["dwi_file"]),
+            name=f"buffer{num_buffers:02}",
+        )
 
     buffernodes.append(get_buffernode())
+
     workflow.connect([
-        (inputnode, buffernodes[-1], [('dwi_file', 'dwi_file')]),
+        # The first buffernode is the raw file
+        (inputnode, buffernodes[0], [('dwi_file', 'dwi_file')]),
+        # XXX: Why pass the bval and bvec files through unmodified?
```

Comment:
The bval and bvec won't be changed by denoising

Reply:
It's hard for me to tell, but it looks like run-wise bval and bvec files get concatenated in `init_merge_and_denoise_wf` only if `denoise_before_combining` is True. It just seems like this could be made clearer through comments.

Reply:
agreed. This whole workflow needs comments